kratos/processes/check_same_modelpart_using_skin_distance_process.cpp

```diff
@@ -0,0 +1,230 @@
+void CheckSameModelPartUsingSkinDistanceProcess<TDim>::Execute()
+{
+    KRATOS_TRY
+
+    // We get the names of the model parts
+    const std::string& r_skin_model_part_1_name = mThisParameters["skin_model_part_1_name"].GetString();
+    const std::string& r_skin_model_part_2_name = mThisParameters["skin_model_part_2_name"].GetString();
+
+    // We get the model parts
+    ModelPart& r_skin_model_part_1 = mrModel.GetModelPart(r_skin_model_part_1_name);
+    ModelPart& r_skin_model_part_2 = mrModel.GetModelPart(r_skin_model_part_2_name);
+    
+    // Checking that the model part contains conditions
+    KRATOS_ERROR_IF(r_skin_model_part_1.NumberOfConditions() == 0) << "The first model part does not contain conditions" << std::endl;
+    KRATOS_ERROR_IF(r_skin_model_part_2.NumberOfConditions() == 0) << "The second model part does not contain conditions" << std::endl;
+    if constexpr (TDim == 2) { // 2D
+        KRATOS_ERROR_IF_NOT(r_skin_model_part_1.Conditions().begin()->GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line2D2) << "Conditions from first model part must be lines in 2D space" << std::endl;
+        KRATOS_ERROR_IF_NOT(r_skin_model_part_2.Conditions().begin()->GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line2D2) << "Conditions from second model part must be lines in 2D space" << std::endl;
+    } else { // 3D
+        KRATOS_ERROR_IF_NOT(r_skin_model_part_1.Conditions().begin()->GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle3D3) << "Conditions from first model part must be triangles in 3D space" << std::endl;
+        KRATOS_ERROR_IF_NOT(r_skin_model_part_2.Conditions().begin()->GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle3D3) << "Conditions from second model part must be triangles in 3D space" << std::endl;
+    }
+
+    // We get the coordinates of the bounding box
+    const double zero_tolerance = std::numeric_limits<double>::epsilon();
+    using NodeType = Node<3>;
+    Parameters mesher_parameters(R"({
+        "number_of_divisions"        : -1,
+        "element_name"               : "Element3D4N",
+        "create_skin_sub_model_part" : false
+    })");
+    mesher_parameters["number_of_divisions"].SetInt(mThisParameters["number_of_divisions_background_mesh"].GetInt());
+    ModelPart& r_model_part_1 = mrModel.CreateModelPart("BACKGROUND_MESH_1");
+    const double bounding_box_scale_factor = mThisParameters["bounding_box_scale_factor"].GetDouble();
+    if constexpr (TDim == 2) { // 2D
+        mesher_parameters["element_name"].SetString("Element2D3N");
+        using BBReduction = CombinedReduction<MaxReduction<double>, MinReduction<double>, MaxReduction<double>, MinReduction<double>>;
+        double max_x, min_x, max_y, min_y;
+        std::tie(max_x, min_x, max_y, min_y) = block_for_each<BBReduction>(r_skin_model_part_1.Nodes(), [&](NodeType& rNode) {
```

Comment:
can you please try to use structured bindings here?

```suggestion
        auto [max_x, min_x, max_y, min_y] = block_for_each<BBReduction>(r_skin_model_part_1.Nodes(), [&](NodeType& rNode) {
```

Reply:
Doesn't work

Reply:
Hm thought this would work 🤔 
I am curious, can you please post the error?

Reply:
Now look like it works, I will do a minor PR, sorry for the noise